### PR TITLE
TRUNK-5564:PersonAddressValidator - Throws exception instead of logging

### PR DESCRIPTION
**Pull request by jwnasambu**
## Description of what I changed
I coded the PersonAddressValidator to avoids throwing exceptions

## Issue I worked on
https://issues.openmrs.org/browse/TRUNK-5564

### CODE REVIEW

**Comment by jwnasambu** [comment]
@dkayiwa kindly review my PR at your convenient time. Thank you

File: api/src/main/java/org/openmrs/validator/PersonAddressValidator.java [hunk 1]
```diff
@@ -62,14 +62,14 @@ public void validate(Object object, Errors errors) {
...
-		}
-		
 		PersonAddress personAddress = (PersonAddress) object;
-		
-		//resolve a shorter name to display along with the error message
+//resolve a shorter name to display along with the error message
+		if (object == null) {
+			 errors.reject("error.address");
+		 } else if (!personAddress.getVoided()) {
+		// TODO - the following method should be made private in a major release
```
Comment (gitcliff) [inline api/src/main/java/org/openmrs/validator/PersonAddressValidator.java]: @jwamalwa can you remove the TODO comments 
https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-logging

File: api/src/main/java/org/openmrs/validator/PersonAddressValidator.java [hunk 2]
```diff
@@ -131,4 +131,9 @@ public void validate(Object object, Errors errors) {
 		    "address5", "address6", "address7", "address8", "address9", "address10", "address11", "address12", "address13", 
 		    "address14", "address15");
 	}
+
+	private void validatePersonAddress(PersonAddress personAddress, Errors errors, boolean b, boolean c) {
+		// TODO Auto-generated method stub
```
Comment (gitcliff) [inline api/src/main/java/org/openmrs/validator/PersonAddressValidator.java]: also here

File: api/src/main/java/org/openmrs/validator/PersonAddressValidator.java [hunk 3]
```diff
@@ -62,14 +62,13 @@ public void validate(Object object, Errors errors) {
...
-		
 		PersonAddress personAddress = (PersonAddress) object;
-		
-		//resolve a shorter name to display along with the error message
+//resolve a shorter name to display along with the error message
+		if (object == null) {
+			 errors.reject("error.address");
+		 } else if (!personAddress.getVoided()) {
+			validatePersonAddress(personAddress, errors, false, true);
+					}
```
Comment (dilanthas) [inline api/src/main/java/org/openmrs/validator/PersonAddressValidator.java]: I think this is not formatted properly

**Pull request closed without merging**